rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol

```diff
@@ -258,20 +257,7 @@ contract RBTCWrapperProxy is ContractRegistryClient {
         _poolToken.approve(address(liquidityMiningContract), poolTokenAmount);
         liquidityMiningContract.deposit(address(_poolToken), poolTokenAmount, msg.sender);
 
-        for (uint256 i = 1; i < _reserveTokens.length; i++) {
-            amount = _reserveAmounts[i].sub(ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).fundCost(totalSupplyBefore, rsvBalances[i], reserveRatio_, poolTokenAmount));
-            if (amount > 0) {
-                success = _reserveTokens[i].transfer(msg.sender, amount);
-                require(success, "Failed to transfer extra reserve token back to user");
-            }
-        }
-        
-        amount = _reserveAmounts[0].sub(ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).fundCost(totalSupplyBefore, rsvBalances[0], reserveRatio_, poolTokenAmount));
-        if (amount > 0) {
-            IWrbtcERC20(wrbtcTokenAddress).withdraw(amount);
-            (success,) = msg.sender.call.value(amount)("");
-            require(success, "Failed to send extra RBTC back to user");
-        }
+        _refund(_liquidityPoolConverterAddress, _reserveTokens, _reserveAmounts, rsvBalances, totalSupplyBefore, poolTokenAmount, reserveRatio_);
```

Comment:
i think the refund logoc is too complicated and consumes too much gas. it would probably be easier and more gas efficient to compare the contracts own token balances and see if `oldBalance - reserveAmounts[i] < newBalance` and refund the difference (instead of querying the reserve ratios and sovryn swap formula several times).

Reply:
i know this is an improvement of existing code and directly related to your task, but because we're already touching this contract, it would be good to take the opportunity to reduce the gas costs.

Reply:
Please also compare gas before and gas after

Reply:
Updated the refund logic.
The gas comparison:
4 add liquidity txs for [The previous version](https://github.com/DistributedCollective/oracle-based-amm/blob/c7787db3f6a2fe45a0ba516bd5d42f7b3c98f95d/rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol#L458-L480) , the average gas consumption is 302498

4 add liquidity txs for [The updated version](https://github.com/DistributedCollective/oracle-based-amm/blob/863c9c7a3a4eff50b12f15fd834fbc25c948c2c1/rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol#L454-L474) , the average gas consumption is 295849

The advantages of the updated version are: 
1. less gas consumption 
2. Code is more readable since we don't need to get the pool's ratio, total supply before & after, etc.
3. Have more stack space to avoid the stack too deep issue.